Source/UnitTests/Common/SettingsHandlerTest.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <gtest/gtest.h>
+
+#include "Common/SettingsHandler.h"
+
+namespace
+{
+constexpr size_t BUFFER_SIZE = Common::SettingsHandler::SETTINGS_SIZE;
+
+// The encrypted bytes corresponding to the following settings, in order:
+//   "key" = "val"
+std::array<u8, 9> BUFFER_CONTENTS_A = {(u8)'\x91', (u8)'\x91', (u8)'\x90', (u8)'\xEE', (u8)'\xD1',
+                                       (u8)'/',    (u8)'\xF0', (u8)'4',    (u8)'y'};
+
+// The encrypted bytes corresponding to the following settings, in order:
+//   "key1" = "val1"
+//   "key2" = "val2"
+//   "foo" = "bar"
+std::array<u8, 31> BUFFER_CONTENTS_B = {
+    (u8)'\x91', (u8)'\x91', (u8)'\x90', (u8)'\xE2', (u8)'\x9A', (u8)'8', (u8)'\xFD', (u8)'U',
+    (u8)'B',    (u8)'\xEA', (u8)'\xC4', (u8)'\xF6', (u8)'^',    (u8)'\xF', (u8)'\xDF', (u8)'\xE7',
+    (u8)'\xC3', (u8)'\n',   (u8)'\xBB', (u8)'\x9C', (u8)'P',    (u8)'\xB1', (u8)'\x10', (u8)'\x82',
+    (u8)'\xB4', (u8)'\x8A', (u8)'\r',   (u8)'\xBE', (u8)'\xCD', (u8)'r',    (u8)'\xF4'};
+}
+
+TEST(SettingsHandlerTest, EncryptSingleSetting)
+{
+  Common::SettingsHandler handler;
+  handler.AddSetting("key", "val");
+  Common::SettingsHandler::Buffer buffer = handler.GetBytes();
+
+  ASSERT_EQ(buffer.size(), BUFFER_SIZE);
+  for (size_t i = 0; i < BUFFER_CONTENTS_A.size(); ++i)
+  {
+    EXPECT_EQ(buffer[i], BUFFER_CONTENTS_A[i]);
+  }
```

Comment:
This loop could be replaced with calling `std::equal` or `std::memcpy`. Same for the other tests.

Reply:
Done (here and below).